tests/Unit/Evolution/Initialization/Test_FunctionOfTimeFromFile.cpp

```diff
@@ -0,0 +1,138 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/ElementId.hpp"
+#include "Domain/ElementIndex.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/FunctionsOfTime/Tags.hpp"
+#include "Evolution/Initialization/FunctionOfTimeFromFile.hpp"
+#include "Evolution/Initialization/Tags.hpp"
+#include "Informer/InfoFromBuild.hpp"
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "Parallel/PhaseDependentActionList.hpp"
+#include "Utilities/FileSystem.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "tests/Unit/ActionTesting.hpp"
+
+namespace {
+constexpr size_t dim = 2;
+using ElementIndexType = ElementIndex<dim>;
+
+template <typename Metavariables>
+struct component {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementIndexType;
+  using simple_tags = tmpl::list<Initialization::Tags::FuncOfTimeFile,
+                                 Initialization::Tags::FuncOfTimeSetNames>;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Initialization,
+          tmpl::list<ActionTesting::InitializeDataBox<simple_tags>>>,
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Testing,
+          tmpl::list<Initialization::Actions::FunctionOfTimeFromFile>>>;
+};
+
+struct Metavariables {
+  using component_list = tmpl::list<component<Metavariables>>;
+  enum class Phase { Initialization, Testing, Exit };
+};
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.FunctionOfTimeFromFile",
+                  "[Unit][Evolution][Actions]") {
+  using MockRuntimeSystem = ActionTesting::MockRuntimeSystem<Metavariables>;
+
+  const ElementId<dim> self_id(1);
+
+  MockRuntimeSystem runner{{}};
+  ActionTesting::emplace_component_and_initialize<component<Metavariables>>(
+      &runner, self_id,
+      {db::item_type<Initialization::Tags::FuncOfTimeFile>{
```

Comment:
Please write out the type

Reply:
Sure! I got misled by the test I based this off of, to learn how the action framework works (can't remember which one) that did this